### PR TITLE
Expose contact point in DCAT RDF

### PR DESCRIPTION
**Pull request by maudetes**
Close https://github.com/datagouv/data.gouv.fr/issues/1520
Close https://github.com/datagouv/data.gouv.fr/issues/1519

Example:
```
  <rdf:Description rdf:about="http://dev.local:7000/dataservices/6661e3b3ac0706063080aa40/">
    <rdf:type rdf:resource="http://www.w3.org/ns/dcat#DataService"/>
    <dcat:contactPoint rdf:resource="http://dev.local:7000/api/1/contacts/6661de3080af2e6f478a53f7/"/>
  </rdf:Description>
  <rdf:Description rdf:about="http://dev.local:7000/api/1/contacts/6661de3080af2e6f478a53f7/">
    <rdf:type rdf:resource="http://www.w3.org/2006/vcard/ns#Kind"/>
    <vcard:fn>Support API Météo-France</vcard:fn>
    <vcard:hasEmail rdf:resource="mailto:contact.api@meteo.fr"/>
    <vcard:hasUrl rdf:resource="https://ceci.est.un.test.com/"/>
  </rdf:Description>
```

### CODE REVIEW

File: udata/core/dataservices/rdf.py [hunk 1]
```diff
@@ -116,10 +117,10 @@ def dataservice_to_rdf(dataservice: Dataservice, graph=None):
     d.set(DCT.issued, Literal(dataservice.created_at))
 
     if dataservice.base_api_url:
-        d.set(DCAT.endpointURL, Literal(dataservice.base_api_url))
+        d.set(DCAT.endpointURL, URIRef(dataservice.base_api_url))
```
Comment (maudetes) [inline udata/core/dataservices/rdf.py:120]: dataservice endpointURL and endpointDescription are actually URLs and are now exposed as URIRef (same as other URLs in RDF, ex accessUrl and downloadUrl)

File: udata/rdf.py [hunk 2]
```diff
@@ -301,22 +302,64 @@ def contact_point_from_rdf(rdf, dataset):
             or rdf_value(contact_point, VCARD.email)
             or rdf_value(contact_point, DCAT.email)
         )
-        if not email:
-            return
         email = email.replace("mailto:", "").strip()
+        contact_form = rdf_value(contact_point, VCARD.hasUrl)
+        if not email and not contact_form:
+            return
```
Comment (maudetes) [inline udata/rdf.py:308]: We're now requiring either an email OR a contact_form to have a proper contact point.

File: udata/rdf.py [hunk 3]
```diff
@@ -301,22 +302,64 @@ def contact_point_from_rdf(rdf, dataset):
...
             ).first()
             return (
                 contact_point
-                or ContactPoint(name=name, email=email, organization=dataset.organization).save()
+                or ContactPoint(
+                    name=name,
+                    email=email,
+                    contact_form=contact_form,
+                    organization=dataset.organization,
+                ).save()
```
Comment (magopian) [inline udata/rdf.py:320]: Nit: would it make sense to use `udata.mongo.queryset.UDataQuerySet.get_or_create` for such use cases here and below with the owner?

**Review by magopian — APPROVED** [review]
Nice!
A couple of nits that you may ignore ;)

Thanks for pairing on that piece of functionality, it helped me greatly getting to know the inner workings of it, and where to look for specifications and examples 🙏

File: udata/rdf.py (hunk 3 above)
Comment (maudetes) [reply udata/rdf.py]: :+1:  Indeed! Done in https://github.com/opendatateam/udata/pull/3179/commits/75091ce7d009fa5049ca827eefb77f375933169b

File: udata/tests/dataset/test_dataset_rdf.py [hunk 4]
```diff
@@ -116,6 +124,11 @@ def test_all_dataset_fields(self):
         org = d.value(DCT.publisher)
         assert org.value(RDF.type).identifier == FOAF.Organization
         assert org.value(FOAF.name) == Literal("organization")
+        c = d.value(DCAT.contactPoint)
```
[the remark this answers is not in the document]
Comment (maudetes) [reply udata/tests/dataset/test_dataset_rdf.py]: Done in https://github.com/opendatateam/udata/pull/3179/commits/82680229c816996936aa20c69ee89e365fd79490

**Pull request merged**